### DISCUSSION TITLE
MQTT control of pool and spa temperature

### DISCUSSION THREAD

**Discussion by antamy** [tagyoureit/nodejs-poolController#434] [Q&A] [answered]
Has anyone been able to use MQTT to change the setPoint of the spa or pool? 

When I publish eg.

`{"setPoint":95}` 

to 

`intellitouch-i53/state/temps/bodies/2/spa/setPoint`

nothing happens. Yet, when I change the setPoint in the poolController UI I see the same MQTT publish messages being sent to the same topic. I must be missing something!

Anthony

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC to ensure you have the changes I just posted and post the following
```
{ "id": 2, "heatSetpoint":95 }
```
to the topic `intellitouch-i53/state/body/setpoint`.  You cannot both subscribe to a topic and publish it at the same time or you get into a feedback loop.

**Reply by antamy** [reply]
ok, thanks.  I'll give it a try later this evening.  You didn't happen to do the same for the pool did you??

**Reply by rstrouse** [reply]
Yes simply change the id to 1.  1=pool 2=spa.

**Reply by antamy** [reply]
Seems to work, not that I expected otherwise! 

Thanks for the update!
